packages/components/RadioGroup/src/RadioButton.macos.tsx

```diff
@@ -0,0 +1,56 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { ensureNativeComponent } from '@fluentui-react-native/component-cache';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { RadioGroupContext } from './RadioGroup';
+import { radioButtonName, IRadioButtonProps, IRadioButtonSlotProps, IRadioButtonType } from './RadioButton.types';
+
+const NativeRadioButtonView = ensureNativeComponent('RadioButtonView');
+
+export const RadioButton = compose<IRadioButtonType>({
+  displayName: radioButtonName,
+  usePrepareProps: (userProps: IRadioButtonProps, useStyling: IUseComposeStyling<IRadioButtonType>) => {
+    const { content, buttonKey, disabled } = userProps;
+    const info = React.useContext(RadioGroupContext);
+    // Reroute the native component's onPress event to RadioGroup's onChange
+    const onPressRerouted = () => {
+      // Prevent calls to RadioGroup's onChange on the currently selected button
+      if (buttonKey != info.selectedKey) {
+        info.onChange && info.onChange(buttonKey);
+      }
```

Comment:
The extra check is so that if the user clicks the same radio button 100 times, only one `oChange` callback is ever fired? Is that to match the win32 implementation?

Reply:
Correct, that's to match the win32 implementation. I believe that's the macOS behavior as well.